end2end.test/src/test/scala/com/expedia/adaptivealerting/pipeline/test/ConstantThresholdBasedE2ETestSpec.scala

```diff
@@ -0,0 +1,118 @@
+import com.expedia.adaptivealerting.kafka.detector.ConstantThresholdOutlierDetectorStreamRunnerBuilder
+import com.expedia.adaptivealerting.kafka.router.MetricRouterStreamBuilder
+import com.expedia.adaptivealerting.pipeline.{EmbeddedKafka, IntegrationTestSpec}
+import com.expedia.www.haystack.commons.entities.MetricPoint
+import org.apache.kafka.clients.admin.AdminClient
+import org.apache.kafka.streams.KeyValue
+import org.apache.kafka.streams.integration.utils.IntegrationTestUtils
+
+import scala.collection.JavaConverters._
+import scala.concurrent.duration._
+
+class ConstantThresholdBasedE2ETestSpec extends IntegrationTestSpec {
+  protected val INTERMEDIATE_TOPIC = "constant-metrics"
+  protected val metricRouterStreamConfig = streamConfigProperties("metricrouter", "metric-router")
+  protected val constantThresholdStreamConfig = streamConfigProperties("constant-detector", "constant-threshold-detector")
+  protected var constantThresholdTopicConsumerConfig = consumerConfig("constant-detector-consumer", "constant-threshold-detector")
+  protected var anomalyTopicConsumerConfig = consumerConfig("anomalies-consumer","anomaly")
+
+
+  override def beforeEach() {
+    EmbeddedKafka.CLUSTER.createTopic(INPUT_TOPIC)
+    EmbeddedKafka.CLUSTER.createTopic(INTERMEDIATE_TOPIC)
+    EmbeddedKafka.CLUSTER.createTopic(OUTPUT_TOPIC)
+    IntegrationTestUtils.purgeLocalStreamsState(metricRouterStreamConfig)
+    IntegrationTestUtils.purgeLocalStreamsState(constantThresholdStreamConfig)
+  }
+
+  override def afterEach(): Unit = {
+    EmbeddedKafka.CLUSTER.deleteTopic(INPUT_TOPIC)
+    EmbeddedKafka.CLUSTER.deleteTopic(INTERMEDIATE_TOPIC)
+    EmbeddedKafka.CLUSTER.deleteTopic(OUTPUT_TOPIC)
+  }
+
+  "AA with 'Constant Threshold Anomaly Detection' model" should {
+
+    "consume anomalous metrics from input topic, classify and confirm them as anomalies and " +
+      "send results to output topic" in {
+
+      Given("a set of anomalous metrics and kafka specific configurations")
+      val metrics = generateAnomalousMetrics()
+      val metricRouterSR = metricRouterStreamRunner
+      val constantThresholdSR = constantThresholdDetectorStreamRunner
+
+      When("anomalous metrics are produced in 'input' topic, and kafka-streams topology is started")
+      produceSpansAsync(10.millis, metrics)
+      metricRouterSR.start()
+      constantThresholdSR.start()
+
+      Then("'metric router' should route the metrics to 'constant threshold outlier detector' model's input topic")
+
+      val intermediateRecords: List[KeyValue[String, MetricPoint]] =
+        IntegrationTestUtils.waitUntilMinKeyValueRecordsReceived[String, MetricPoint](
+          configToProps(constantThresholdTopicConsumerConfig.getConfig(STREAM)),
+          INTERMEDIATE_TOPIC, 2, 15000).asScala.toList // get metricPoints from Kafka's output topic
+      intermediateRecords.size shouldEqual 2
+      intermediateRecords.foreach(i => {
```

Comment:
you could scala pattern matching here

Reply:
something like  intermediateRecords.foreach (record => record.value.metric match { case "latency" => succeed, case "duration" => succeed,  case _ => fail)